### ISSUE TITLE
TorchTrainer updates weights on every batch during gradient accumulation

### ISSUE THREAD

**Issue by LckyLke** [dice-group/dice-embeddings#432]
With accumulation set to 2, weights should remain 0 after the first batch and become 0.2 after the second. Instead, they become 0.2 and 0.56: optimizer.step() runs every batch and reapplies accumulated gradients.

Reproduced on `66a7e619` :

```python
from types import SimpleNamespace
import torch
from dicee.trainer.torch_trainer import TorchTrainer

trainer = TorchTrainer(SimpleNamespace(random_seed=0, gradient_accumulation_steps=2), [])
trainer.train_dataloaders = [None, None]
w = torch.nn.Parameter(torch.tensor(0.0))
trainer.optimizer = torch.optim.SGD([w], lr=0.1)
trainer.training_step = lambda batch: (w - 1).square()
trainer._run_batch(0, None, None)
print(f'After first batch: {w.item():.2f} (expected 0.00)')
trainer._run_batch(1, None, None)
print(f'After second batch: {w.item():.2f} (expected 0.20)')
assert torch.isclose(w, torch.tensor(0.2))
```
